### DISCUSSION TITLE
jfa-go initial configurations loop

### DISCUSSION THREAD

**Discussion by Ahmed-Hegab** [hrfee/jfa-go#352] [Q&A] [answered]
I deployed jellyfin and jfa-go on the same pod on GKE all work good except the initial configurations of jfa-go loops, after entering all data and submit it tells me to refresh to start, it duplicate the url and give me page not found then after the correction of the url it tells to enter the initial configurations from beginning and so on ..
anyone help ?

**Comment by hrfee** [comment]
Is there any output in the logs? It might be a permissions issue but I'm not usre.

**Reply by Ahmed-Hegab** [reply]
severity: "ERROR"
textPayload: "badger 2024/08/02 18:57:12 INFO: Lifetime L0 stalled for: 0s"

severity: "ERROR"
textPayload: "badger 2024/08/02 18:57:12 INFO: "

textPayload: "Level 0 [ ]: NumTables: 01. Size: 776 B of 0 B. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 64 MiB"
timestamp: "2024-08-02T18:57:12.709447577Z"

severity: "ERROR"
textPayload: "Level 1 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB"
timestamp: "2024-08-02T18:57:12.709460388Z"

severity: "ERROR"
textPayload: "Level 2 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB"
timestamp: "2024-08-02T18:57:12.709465869Z"

severity: "ERROR"
textPayload: "Level 3 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB"
timestamp: "2024-08-02T18:57:12.709470458Z"

severity: "ERROR"
textPayload: "Level 4 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB"
timestamp: "2024-08-02T18:57:12.709475419Z"

severity: "ERROR"
textPayload: "Level 5 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB"
timestamp: "2024-08-02T18:57:12.709480741Z"

severity: "ERROR"
textPayload: "Level 6 [B]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB"
timestamp: "2024-08-02T18:57:12.709487162Z"

severity: "ERROR"
textPayload: "Level Done"
timestamp: "2024-08-02T18:57:12.709492133Z"

this happens only when I press submit button after entering all initial config data, what might be the problem?

**Reply by hrfee** [reply]
Are these raw logs, or is this some processed log output from GKE? These logs appear to be jfa-go shutting down, if it's a crash there should be some more information earlier than this.

**Reply by Ahmed-Hegab** [reply]
textPayload: "[INFO] 18:57:11  Config modification requested"
timestamp: "2024-08-02T18:57:11.166728640Z"

textPayload: "[INFO] 18:57:11  Restarting..."
timestamp: "2024-08-02T18:57:11.177154565Z"

textPayload: "[INFO] 18:57:11  Restart/Quit signal received, give me a second!"
timestamp: "2024-08-02T18:57:11.177208100Z"

textPayload: "[ERROR] 18:57:11 main.go:573: Failure serving: http: Server closed"
timestamp: "2024-08-02T18:57:11.183657893Z"

textPayload: "[GIN] POST(/config) => 200"
timestamp: "2024-08-02T18:57:12.179479530Z"

textPayload: "[INFO] 18:57:12  Server shut down."
timestamp: "2024-08-02T18:57:12.269860513Z"

textPayload: "badger 2024/08/02 18:57:12 INFO: All 0 tables opened in 0s"
timestamp: "2024-08-02T18:57:12.456528437Z"

textPayload: "badger 2024/08/02 18:57:12 INFO: Discard stats nextEmptySlot: 0"
timestamp: "2024-08-02T18:57:12.459634427Z"

textPayload: "badger 2024/08/02 18:57:12 INFO: Set nextTxnTs to 0"
timestamp: "2024-08-02T18:57:12.459671550Z"
}


its GKE logs, there no anything else to tell me what is the problem
it is another sample of log

**Reply by Ahmed-Hegab** [reply]
severity: "INFO"
textPayload: "[INFO] 15:38:37  Authenticated with
timestamp: "2024-08-04T15:38:37.475489750Z"

severity: "INFO"
textPayload: "[INFO] 15:38:37  Migrating to Badger(hold)"
timestamp: "2024-08-04T15:38:37.476959638Z"

textPayload: "[INFO] 15:38:37  All data migrated to database. JSON files in the config folder can be deleted if you are sure all data is correct in the app. Create an issue if you have problems."
timestamp: "2024-08-04T15:38:37.479048798Z"

severity: "INFO"
textPayload: "[INFO] 15:38:37  Initializing router"
timestamp: "2024-08-04T15:38:37.480733277Z"

severity: "INFO"
textPayload: "[INFO] 15:38:37  User daemon started"
timestamp: "2024-08-04T15:38:37.483615451Z"

severity: "INFO"
textPayload: "[INFO] 15:38:37  Starting password reset daemon"
timestamp: "2024-08-04T15:38:37.483656051Z"

severity: "INFO"
textPayload: "[INFO] 15:38:37  Invite daemon started"
timestamp: "2024-08-04T15:38:37.483665988Z"

severity: "INFO"
textPayload: "[INFO] 15:38:37  Loading routes"
timestamp: "2024-08-04T15:38:37.500757619Z"

severity: "ERROR"
textPayload: "badger 2024/08/04 15:38:37 INFO: Lifetime L0 stalled for: 0s"
timestamp: "2024-08-04T15:38:37.503092333Z"

severity: "ERROR"
textPayload: "badger 2024/08/04 15:38:37 INFO: "
timestamp: "2024-08-04T15:38:37.515622612Z"

severity: "ERROR"
textPayload: "Level 0 [ ]: NumTables: 01. Size: 776 B of 0 B. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 64 MiB"
timestamp: "2024-08-04T15:38:37.515682663Z"

severity: "ERROR"
textPayload: "Level 1 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB"
timestamp: "2024-08-04T15:38:37.515709687Z"

textPayload: "wildcards must be named with a non-empty name in path 'ang/:page'"
timestamp: "2024-08-04T15:38:37.525524636Z"

**Reply by hrfee** [reply, marked as answer]
> textPayload: "wildcards must be named with a non-empty name in path '/http:/35.193.6.98:8056/lang/:page'"

Did you change the "URL Base" value on the setup page? that value should be just what comes after the domain/IP, i.e. It should be `/accounts` if you're putting jfa-go on the subfolder `http://jellyf.in/accounts`. If you have jfa-go on a subdomain (i.e. at `accounts.jellyf.in`), then you can leave the value blank.

**Reply by Ahmed-Hegab** [reply]
they are on the same IP Address but everyone on separate port
they are on the same container and same pod

**Reply by Ahmed-Hegab** [reply]
thanks, it working now

**Comment by Ahmed-Hegab** [comment]
after I set the domain and ssl through GKE secret, both jellyfin and jfa-go
work secure with the domain names and jfa-go login to users and
authenticate them coreeclty, but jfa-go do not get the users in accounts
tab and do not send forget password emails, however it was working on http,
is there a specific reason?

‫في الأحد، 4 أغسطس 2024 في 10:55 م تمت كتابة ما يلي بواسطة ‪Harvey
Tindall‬‏ ***@***.***‬‏>:‬

**Reply by hrfee** [reply]
Would you mind having a look in the javascript console in your browser for errors?

**Comment by Ahmed-Hegab** [comment]
everything is good now, invitation emails is working fine, but forget
password emails cannot be send unless the email address is modified from
the accounts page despite I checked the checkbox email address as username
in the setup, is there a solution to send the forget password using
username as the user enter it as valid email ?

‫في الأحد، 11 أغسطس 2024 في 10:07 م تمت كتابة ما يلي بواسطة ‪Harvey
Tindall‬‏ ***@***.***‬‏>:‬

**Reply by hrfee** [reply]
Sorry I don't really understand the issue. What error is stopping password reset/forgot password messages from working? If email isn't listed as an option, make sure "Allow PWR with email address" is enabled in Settings > User Page. If not, i'm not sure, could you check the logs?

**Comment by Ahmed-Hegab** [comment]
It was a version problem, It solved now,
There is something I want to do, If I build from source code and I changed
the config.json to be my initial configs I want, how to skip wizard and
call the config function to write my config.json in config.ini file at the
beginning of the application, without force the user to make the initial
configs, is there a way ?

**Reply by hrfee** [reply]
In `main.go`, the value `firstRun` is set to true (around line 217), and the contents of `config-default.ini` is copied to the config location. To stop setup starting, you can simply set `firstRun = false` after the config has copied. `config-default.ini` is generated from `config/config-base.json`, but you could easily modify it directly. The temporary config value `first_run` is also set true in the new config file around line 236, so that the program won't treat this default config as complete. You can remove this line so the program will treat it as complete.
